app/views/assignments/_form.html.erb

```diff
@@ -18,40 +19,73 @@
   <%= f.file_field :mapfile, accept: '.csv' %>
-
+  </div>
+  <div id="file_to_ignore_details" style="display:none;">
+  <%= f.label :filesIgnore, "Files to be ignored \u24D8" , title: "If you want to ignore certain files from the code zip file that you uploaded, add those filenmes under here. \n\nIf you want to ignore filenames that exactly match a certain filename, enter the filename as it is. For instance, entering \"question1\", will mean that all the filenames that are named as \"question1\" (irrespective of their file extensions) will all be ignored. \n\nIf you want to ignore filenames that partially contain a substring, add \"/ \" in front of the filename. For instance, if you enter, \"/question1\", filenames such as \"alice_question1.py\", \"bob_question1Test.c\", \"question1_charlie.java\" will all get ignored. \n\nIf you need to add muliple regex, leave a space in between them. For instance, if you enter \"question1 /question2 \", filesnames that either exactly match \"question1 \" or contain the substring \"question2 \" will all get ignored"  %>
+  <%= f.text_field :filesIgnore %>
+  </div>
+  <div id="file_to_process_details" style="display:none;">
+  <%= f.label :filesProcess, "Files to be only processed \u24D8" , title: "If you want to process and compare only certain files from the code zip file that you uploaded, add those filenmes under here. \n\nIf you want to process filenames that exactly match a certain filename, enter the filename as it is. For instance, entering \"question1\", will mean that all the filenames that are named as \"question1\" (irrespective of their file extensions) will all be processed and the rest will be ignored. \n\nIf you want to only process filenames that partially contain a substring, add \"/ \" in front of the filename. For instance, if you enter, \"/question1\", filenames such as \"alice_question1.py\", \"bob_question1Test.c\", \"question1_charlie.java\" will all get processed and the rest that does not contain the substring will all get ignored. \n\nIf you need to add muliple regex, leave a space in between them. For instance, if you enter \"question1 /question2 \", filesnames that either exactly match \"question1 \" or contain the substring \"question2 \" will all get processed."  %>
+  <%= f.text_field :filesProcess %>
+  </div>
+  <div id="ref_dir_details" style="display:none;">
+    <%= f.label :refDir, "Reference Directory (zip) \u24D8", title: "If you have additional files that you want to compare, but do not want to check for plagiarism (for example, if you want to also compare submissions from previous semesters), zip those files and upload it here." %>
+    <%= f.file_field :refDir %>
+  </div>
+  <div id="email_details" style="display:none;">
+  <%= f.label :email, "Send email upon completion \u24D8" , title: "If you want to receive an email to notify you after the assignment has been processed, enter the email here. Ensure that you enter a valid email." %>
+  <%= f.text_field :email %>
```

Comment:
Fill in with user's default email, to be edited?

Reply:
Yup, I have changed it accordingly.